### PR TITLE
SWC-7394-search-term-logic

### PR DESCRIPTION
**Pull request by kianamcc**
Relevant Jira: https://sagebionetworks.jira.com/browse/SWC-7394?focusedCommentId=263125&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&sourceType=mention#comment-263125

Previous fix (logic in SynapseHomepageChatSearch) only applied to search on the synapse homepage.

In SWC, when we do a search, the query is a JSON.

Example search in project nav search: syn68899079

We get a query string that looks like this: '%7B%22queryTerm%22%3A%5B%22syn68899101%22%5D%2C%22facetOptions%22%3A%5B%7B%22name%22%3A%22EntityType%22%2C%22maxResultCount%22%3A300%2C%22sortType%22%3A%22COUNT%22%7D%2C%7B%22name%22%3A%22Consortium%22%2C%22maxResultCount%22%3A300%2C%22sortType%22%3A%22COUNT%22%7D%2C%7B%22name%22%3A%22ModifiedOn%22%2C%22maxResultCount%22%3A300%2C%22sortType%22%3A%22COUNT%22%7D%2C%7B%22name%22%3A%22ModifiedBy%22%2C%22maxResultCount%22%3A300%2C%22sortType%22%3A%22COUNT%22%7D%2C%7B%22name%22%3A%22CreatedOn%22%2C%22maxResultCount%22%3A300%2C%22sortType%22%3A%22COUNT%22%7D%2C%7B%22name%22%3A%22Tissue%22%2C%22maxResultCount%22%3A300%2C%22sortType%22%3A%22COUNT%22%7D%2C%7B%22name%22%3A%22CreatedBy%22%2C%22maxResultCount%22%3A300%2C%22sortType%22%3A%22COUNT%22%7D%5D%2C%22start%22%3A0%2C%22size%22%3A30%2C%22booleanQuery%22%3A%5B%7B%22key%22%3A%22node_type%22%2C%22value%22%3A%22project%22%7D%5D%7D'

that is then parsed. queryTerm will look like this: ["syn68899101"]. If we have any spaces, that gets included in the same array via split. Possible solution is to clean up the query terms after parsing the JSON to handle simple text queries and JSON queries.

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/shared/WebConstants.java [hunk 1]
```diff
@@ -73,6 +73,10 @@ public class WebConstants {
     "^" + WIDGET_NAME_REGEX + "+";
   public static final String VALID_ENTITY_ID_REGEX =
     "^[Ss]{1}[Yy]{1}[Nn]{1}\\d+";
+  // Regex that matches Synapse entity ID with optional version (e.g., syn123 or syn123.4)
+  // Equivalent to client-side SYNAPSE_ENTITY_ID_REGEX = /^(syn\d+)(?:\.(\d+))?$/i
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/shared/WebConstants.java]: without the 'i' (since case is not ignored in this regex)

File: src/test/java/org/sagebionetworks/web/unitclient/presenter/SearchPresenterTest.java [hunk 2]
```diff
@@ -453,7 +453,10 @@ public void testSetPlaceUsernamePrefix() throws Exception {
     String userName = "VaderLabTech";
     Place redirectPlace = SearchUtil.willRedirect("@" + userName);
     assertTrue(redirectPlace instanceof PeopleSearch);
-    assertEquals(userName, ((PeopleSearch) redirectPlace).getSearchTerm());
+    assertEquals(
+      userName.toLowerCase(),
+      ((PeopleSearch) redirectPlace).getSearchTerm()
+    );
   }
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/presenter/SearchPresenterTest.java:507]: Consider adding a test where the search terms contain whitespace only, and terms that are surrounded by whitespace (and verify terms are cleaned up).
As well as a new test that is similar to testSetPlaceSynapseIdPrefix(), but also includes a dot version suffix.

**Review by jay-hodgson — APPROVED** [review]
